pkg/build/inline.go

```diff
@@ -138,41 +194,95 @@ func (n *Inliner) ComponentFiles(ctx context.Context, comp *bundle.ComponentBuil
 				}
 				obj, err := converter.FromYAMLString(s).ToUnstructured()
 				if err != nil {
-					return nil, fmt.Errorf("converting multi-doc object number %d for component %q, %v", i, name, err)
+					return nil, fmt.Errorf("converting multi-doc object number %d for component %v, %v", i, ref, err)
 				}
 				annot := obj.GetAnnotations()
 				if annot == nil {
 					annot = make(map[string]string)
 				}
-				annot[string(bundle.InlineTypeIdentifier)] = string(bundle.KubeObjectInline)
 				obj.SetAnnotations(annot)
 				newObjs = append(newObjs, obj)
 			}
 		} else {
 			obj, err := converter.FromFileName(cf.URL, contents).ToUnstructured()
 			if err != nil {
-				return nil, fmt.Errorf("for component %q, %v", name, err)
+				return nil, fmt.Errorf("for component %q, %v", ref, err)
 			}
 			annot := obj.GetAnnotations()
 			if annot == nil {
 				annot = make(map[string]string)
 			}
-			annot[string(bundle.InlineTypeIdentifier)] = string(bundle.KubeObjectInline)
 			obj.SetAnnotations(annot)
 			newObjs = append(newObjs, obj)
 		}
 	}
+	return newObjs, nil
+}
+
+// objectTemplateBuilders builds ObjectTemplates from ObjectTemplateBuilders
+func (n *Inliner) objectTemplateBuilders(ctx context.Context, objects []*unstructured.Unstructured, ref bundle.ComponentReference) ([]*unstructured.Unstructured, error) {
+	var outObj []*unstructured.Unstructured
+	for _, obj := range objects {
+		if obj.GetKind() != "ObjectTemplateBuilder" {
+			outObj = append(outObj, obj)
+			continue
+		}
+		name := obj.GetName()
+
+		builder := &bundle.ObjectTemplateBuilder{}
+		if err := converter.FromUnstructured(obj).ToObject(builder); err != nil {
+			return nil, fmt.Errorf("for component %v and object %q: %v", ref, name, err)
+		}
+
+		contents, err := n.readFile(ctx, builder.TemplateFile)
```

Comment:
Preexisting: Relative paths in config files should probably be interpreted relative to the config file, not the current working directory.

Reply:
Yeah, I need to fix that